sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -283,6 +287,24 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 					newIntegrationSet.Add(ii.GetId())
 				}
 			}
+
+			if env.ForceLocalImageScanning.BooleanSetting() {
```

Comment:
how come this is only done when `saName == ""`?

Reply:
I believe the comments that exist already describe that to an extent, perhaps I can be more verbose. TL;DR; thats the only part that changes with this new feature, storing builder, deployer, etc. secrets will overwrite the entries inserted by the 'defaultSA' block above. So it doens't make sense - the TODO: comment describes the limitation and how it will be changed in the future (with the ticket you had me add as well)

Reply:
Added more comments